### PR TITLE
DBZ-1431

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-1431

@jpechane, as discussed.

### CODE REVIEW

File: kinesis/src/main/java/io/debezium/examples/kinesis/ChangeDataSender.java [hunk 1]
```diff
@@ -85,20 +87,39 @@ public void run() {
...
         catch (InterruptedException e) {
-            Thread.interrupted();
+            Thread.currentThread().interrupt();
         }
     }
 
+    private void cleanUp() {
+        // Lingering MySQL thread without this...
+        try {
+            AbandonedConnectionCleanupThread.shutdown();
```
Comment (jpechane) [inline kinesis/src/main/java/io/debezium/examples/kinesis/ChangeDataSender.java]: Why is this needed? The thread in question is daemon thread. If the thread does not stop then there must be another thread lingering. Also the method in question is deprecated.

Comment (gunnarmorling) [reply kinesis/src/main/java/io/debezium/examples/kinesis/ChangeDataSender.java]: @jpechane, try to remove it, the thread will stick around. Happy to change it, if there's better ways to do it.

Comment (jpechane) [reply kinesis/src/main/java/io/debezium/examples/kinesis/ChangeDataSender.java]: @gunnarmorling I've just checked it for NATS - by the end of shutdown hook only daemon threads are alive (one of them is the MySQL dameon thread). hence I believe it is unnecessary. Otherwise you'd be required to excute the very same operation for any application that uses MySQL JDBC driver.

**Comment by gunnarmorling** [comment]
Can you check with Kinesis, too (just comment out the actual handling logic
and have it raise an exception). Perhaps different driver versions are
involved and later ones don't expose it anymore, so updating the other
examples might help. For that one, without this call above, a thread named
"Abandoned connection cleanup thread" was sticking around.

**Comment by jpechane** [comment]
@gunnarmorling Just checked the code of MySQL connector in Kinesis and the thread there is non-daemon. So I propose to either promote all three examples to `0.9.5.Final` or wait till `0.10.0.Final` is available and promote it to that one.

**Comment by gunnarmorling** [comment]
@jpechane, reworked to use 0.9.5.Final consistently, removing the call for MySQL.

File: nats/README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ This demo shows how to stream changes from MySQL database running on a local mac
 We will start a pre-populated MySQL database that is the same as used by the Debezium [tutorial](http://debezium.io/docs/tutorial/):
 
 ```
-docker run -it --rm --name mysql -p 3306:3306 -e MYSQL_ROOT_PASSWORD=debezium -e MYSQL_USER=mysqluser -e MYSQL_PASSWORD=mysqlpw debezium/example-mysql:0.9
+docker run -it --rm --name mysql -p 3306:3306 -e MYSQL_ROOT_PASSWORD=debezium -e MYSQL_USER=mysqluser -e MYSQL_PASSWORD=mysqlpw debezium/example-mysql:0.10
```
Comment (jpechane) [inline nats/README.md]: @gunnarmorling Why `0.10` is used here and `0.9` in other cases?

Comment (gunnarmorling) [reply nats/README.md]: This was an oversight, good catch. Updated to 0.9.

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks
